pkg/virt-handler/notify-server/server.go

```diff
@@ -84,7 +84,7 @@ func (n *Notify) HandleDomainEvent(ctx context.Context, request *notifyv1.Domain
 	case string(watch.Deleted):
 		n.EventChan <- watch.Event{Type: watch.Deleted, Object: domain}
 	case string(watch.Error):
-		n.EventChan <- watch.Event{Type: watch.Error, Object: status}
```

Comment:
Would it make sense to consider a client side change instead to reduce errors to things we actually care about instead?

For example, if a libvirt event connection is reset, then that's a legitimate error the server should be aware of. But all these other things we're sending errors for, like "domain not found", probably shouldn't have been an error event to begin with.

@rmohr we need to prioritize your domain notify replacement at some point. I'm realizing more and more all the issues doing this serverside in virt-handler involves.

Reply:
I am bit confused by the comment here.
Why would the "server" reset all connections with all virt-launchers due to a single client?
If I understand correctly, the only action taken by the virt-handler in such a case, is to reset the connections and nothing else. If the virt-handler wants to do something else, I guess a mechanism needs to be placed that changes the existing anyway.

So are you suggesting the current behavior is good or that we need to react to errors from the client side?
This change, AFAIU only makes sure we do not reset all connection if a client sent an error. Is this part ok?

Reply:
Yep if i understand right, there is no need to reset the gRPC medium which serves all the client, as Roman said here for example:
https://github.com/kubevirt/kubevirt/pull/3713#issuecomment-652261724

Reply:
the error makes the informer re-list/watch, pulling the latest state.

if you don't process errors, then the informer does not pull the latest state. This means if a libvirt reconnect occurs, the informer might not be accurate if anything changed during the time the connection to libvirt is being re-established.

Now, there's zero reason to recreate the grpc server when an error comes in. that's just a result of how the domain notify list/watch is implemented.

Reply:
Thanks, even that, we need just for some error cases or for all?
will learn if its safe to remove the Stop, in case the watcher will use the same interface again (according a comment there it doesnt, but i will try to check)

Reply:
> the error makes the informer re-list/watch, pulling the latest state.
> 
> if you don't process errors, then the informer does not pull the latest state. This means if a libvirt reconnect occurs, the informer might not be accurate if anything changed during the time the connection to libvirt is being re-established.
> 
> Now, there's zero reason to recreate the grpc server when an error comes in. that's just a result of how the domain notify list/watch is implemented.

Are you saying here that we should not ignore the errors coming in but have an action of refreshing the state?
If this is the case, then this PR merits a nack.
Could you please clarify this?

Reply:
nack in the current state, a little improvement would fix it according what i understood (and according 
https://github.com/kubevirt/kubevirt/pull/3806#issuecomment-657806368)

you can remove the lgtm already if you want, thanks

Reply:
Updated PR, please take another look